application/models/Question.php

```diff
@@ -292,14 +292,16 @@ public function getAdvancedSettingsWithValues($iQuestionID, $sQuestionType, $iSu
             $aLanguages = array($sLanguage);
         }
         $aAttributeValues = QuestionAttribute::model()->getQuestionAttributes($iQuestionID, $sLanguage);
+
         // TODO: move getQuestionAttributesSettings() to QuestionAttribute model to avoid code duplication
         $aAttributeNames = QuestionAttribute::getQuestionAttributesSettings($sQuestionType);
 
         // If the question has a custom template, we first check if it provides custom attributes
-
         $oQuestion = Question::model()->find(array('condition'=>'qid=:qid', 'params'=>array(':qid'=>$iQuestionID)));
         $aAttributeNames = self::getQuestionTemplateAttributes($aAttributeNames, $aAttributeValues, $oQuestion);
-
+        // Add the questions attributes by plugins
+        $aAttributeNames = array_merge($aAttributeNames, QuestionAttribute::getQuestionAttributesPlugins($sQuestionType));
```

Comment:
Why can't this line be removed if getQuestionAttributesPlugins is already called in getQuestionAttributes above?

Reply:
One time for definition, one time for values …

Reply:
But getQuestionAttributes already collect values here:
https://github.com/LimeSurvey/LimeSurvey/blob/master/application/models/QuestionAttribute.php#L240